contracts/GnosisSafe.sol

```diff
@@ -138,14 +138,14 @@ contract GnosisSafe is MasterCopy, ModuleManager, OwnerManager, SignatureDecoder
     )
         private
     {
-        uint256 amount = gasUsed.add(baseGas).mul(gasPrice);
         // solium-disable-next-line security/no-tx-origin
         address payable receiver = refundReceiver == address(0) ? tx.origin : refundReceiver;
         if (gasToken == address(0)) {
+            // For ETH we will only adjust the gas price to not be higher than the actual used gas price
             // solium-disable-next-line security/no-send
-            require(receiver.send(amount), "Could not pay gas costs with ether");
+            require(receiver.send(gasUsed.add(baseGas).mul(gasPrice < tx.gasprice ? gasPrice : tx.gasprice)), "Could not pay gas costs with ether");
```

Comment:
I understand this will make txs more fair for users setting a higher `gasPrice` than the one that was used, but for users setting a lower `gasPrice` it's not possible as they have already calculated the refund, right?

For Safes not created using the `PaymentProxy` it will apply too, as they use this method. Sounds good to me

Reply:
> but for users setting a lower gasPrice it's not possible as they have already calculated the refund

You want to make it fair for us, right? The simplest way for this would be a gas price oracle which the user needs to trust. But I wouldn't include this in the core contracts

Reply:
Agree, it would be very complex. As long as we are confortable making it more fair for the user than for us this is a good solution